app/src/main/java/ch/epfl/cs311/wanderwave/model/auth/AuthenticationController.kt

```diff
@@ -74,28 +80,40 @@ constructor(
   }
 
   private suspend fun storeAndUseNewTokens(responseJson: String): Boolean {
-    val response = JSONObject(responseJson)
-    val firebaseToken = response.getString("firebase_token")
-    val spotifyAccessToken = response.getString("access_token")
-    val spotifyRefreshToken = response.getString("refresh_token")
-
-    tokenRepository.setAuthToken(
-        AuthTokenRepository.AuthTokenType.SPOTIFY_ACCESS_TOKEN,
-        spotifyAccessToken,
-        System.currentTimeMillis() / 1000L + 3600)
-    tokenRepository.setAuthToken(
-        AuthTokenRepository.AuthTokenType.SPOTIFY_REFRESH_TOKEN,
-        spotifyRefreshToken,
-        System.currentTimeMillis() / 1000L + 3600 * 100000)
-    tokenRepository.setAuthToken(
-        AuthTokenRepository.AuthTokenType.FIREBASE_TOKEN,
-        firebaseToken,
-        System.currentTimeMillis() / 1000L + 3600)
-
-    return signInWithCustomToken()
+    if (responseJson.isEmpty()) {
+      Log.d("AuthenticationController", "Received empty JSON response")
+      return false
+    }
+
+    return try {
+      val response = JSONObject(responseJson)
+      val firebaseToken = response.getString("firebase_token")
+      val spotifyAccessToken = response.getString("access_token")
+      val spotifyRefreshToken = response.getString("refresh_token")
+
+      Log.d("AuthenticationController", "SPOTIFY Access token: $spotifyAccessToken")
```

Comment:
Is it a good thing to print this ? Seems like a security issue. I know it's just debug, but we should probably get rid of it now before we forget.

Reply:
You're right, I forgot they were there